### PR TITLE
fix(string): cleanup code in String.prototype.repeat

### PR DESCRIPTION
**Pull request by ghost**
Overview
This PR refactors the String.prototype.repeat implementation to strictly align with Section 22.1.3.16 of the ECMA-262 specification. The previous implementation used a generic error handling match; this update introduces explicit checks for Infinity and negative values as required by the abstract operations.

Spec Compliance Audit
I have mapped the implementation to the following specification steps:

Step 3: Let n be ? ToIntegerOrInfinity(count).

Step 4: If n < 0, throw a RangeError exception. (Implemented via explicit match)

Step 5: If n is +∞, throw a RangeError exception. (Implemented via PositiveInfinity check)

Changes
Replaced the catch-all error handling with an exhaustive match on the IntegerOrInfinity enum.

Added specific RangeError messages that distinguish between negative counts and infinite counts.

Used variable shadowing to simplify the logic flow from an Enum to a primitive i64 after validation.

Verification
I have verified the fix using a local test script across the following edge cases:

JavaScript
"abc".repeat(0);        // Returns "" (Success)
"abc".repeat(-1);       // Throws RangeError: count must be non-negative (Success)
"abc".repeat(Infinity); // Throws RangeError: count must be less than infinity (Success)
Note to Maintainers
I am a prospective GSoC 2026 applicant interested in the ECMAScript Conformance project. I chose this task to demonstrate my ability to translate ECMA-262 algorithms into idiomatic Rust using Boa's internal types. I am eager to contribute more to the engine's compliance suite!

### CODE REVIEW

**Comment by ghost** [comment]
Hi @jedel1043 ! I hope you had a great weekend!
Just a gentle ping on this PR and my other open one (#5018). It looks like both are waiting for a maintainer to approve the GitHub Actions CI workflows. Could you please trigger the checks for them when you have a moment?
I'm standing by to make any updates or fix any formatting issues once the CI runs. Thank you!

File: core/engine/src/builtins/string/mod.rs [hunk 1]
```diff
@@ -689,41 +689,63 @@ impl String {
...
 
-                // Charge each repetition against the VM loop-iteration limit.
-                let mut result = Vec::with_capacity(n);
-                for _ in 0..n {
-                    crate::vm::opcode::IncrementLoopIteration::operation((), context)?;
-                    result.push(string.as_str());
-                }
+        if string.is_empty() {
+            return Ok(js_string!().into());
+        }
```
Comment (jedel1043) [inline core/engine/src/builtins/string/mod.rs]:
```suggestion
        if n == 0 || string.is_empty() {
            return Ok(js_string!().into());
        }
```

**Review by jedel1043 — APPROVED** [review]
Looks great!

**Comment by ghost** [comment]
Hey @jedel1043, I pulled down your web edits, fixed the minor syntax/doc comment error, ran cargo fmt, and pushed it back up. The CI checks are all green and good to go!

**Comment by jedel1043** [comment]
yeah that's what I get for trying to fix this through my phone 🤣🤣🤣
Thank you for the fix!

**Comment by ghost** [comment]
Haha, we’ve all been there @jedel1043 ! Honestly, I’m just glad I could help get it across the finish line. It’s been a great exercise in learning how Boa maps the ECMA-262 steps to Rust.

Now that these are merged, I’m looking to help out with some of the more complex built-ins for the v1.0.0 milestone. Are there any particular areas (maybe in Array or Object) where the team feels the logic is currently the most 'messy' or out of sync with the spec? I’m happy to dig into some of the heavier lifting next!
